src/main/java/Duke.java

```diff
@@ -1,10 +1,141 @@
+import java.io.IOException;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    /*
+        Alas - OOP, using a sledgehammer to crack a nut.
+        Yet another assignment with disproportionate workload #NUSLife
+    */
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException ex) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
     }
+
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+            } catch (DukeException e) {
+                ui.showError(e.getMessage());
+            } finally {
+                ui.showLine();
+            }
+        }
+    }
+
+    public static void main(String[] args) throws DukeException, IOException {
+        new Duke("list.txt").run();
+    }
+
```

Comment:
Maybe it is advisable to remove the commented code if it is not used to avoid confusion.

Reply:
You caught me mid-refactor, laptop ran out of battery so I had to make a Checkpoint commit.